src/toxcore/tcp/codec.rs

```diff
@@ -109,13 +111,10 @@ impl Decoder for Codec {
     fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
         // deserialize EncryptedPacket
         let (consumed, encrypted_packet) = match EncryptedPacket::from_bytes(buf) {
-            IResult::Incomplete(_) => {
-                return Ok(None)
-            },
-            IResult::Error(error) => {
-                return Err(DecodeError::DeserializeEncryptedError { error, buf: buf.to_vec() })
-            },
-            IResult::Done(i, encrypted_packet) => {
+            Err(Err::Incomplete(_)) => return Ok(None),
+            Err(Err::Error(_)) => return Ok(None),
+            Err(Err::Failure(_)) => return Ok(None),
```

Comment:
Why ok?

Reply:
If it is not ok then this happens

`toxcore::tcp::client::client::tests::spawn ... test toxcore::tcp::client::client::tests::spawn has been running for over 60 seconds`

But I can't resolve it why test hangs.